kernel/kernel-api/src/main/java/io/delta/kernel/internal/TableBuilderImpl.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.kernel.internal;
+
+import static java.util.Objects.requireNonNull;
+
+import io.delta.kernel.Table;
+import io.delta.kernel.TableBuilder;
+import io.delta.kernel.TableIdentifier;
+import io.delta.kernel.config.ConfigurationProvider;
+import io.delta.kernel.engine.Engine;
+import io.delta.kernel.internal.util.Clock;
+import java.util.Optional;
+
+public class TableBuilderImpl implements TableBuilder {
+  private final Engine engine;
+  private final String path;
+
+  private Optional<ConfigurationProvider> configProviderOpt = Optional.empty();
+  private Optional<TableIdentifier> tableIdOpt = Optional.empty();
+
+  /** Note: This is not for a public API. It's for internal testing use only. */
+  private Optional<Clock> clockOpt = Optional.empty();
```

Comment:
There's no need to add `opt` suffix in my opinion. It's obvious from the type.

Reply:
I hear that. This is certainly one of my personal preferences. I like variables being fully self-describing. The value isn't a `Clock`, but distinctly a `Clock Optional`. With this current naming convention, if we did a `.map`, I could refer to the inner value as `clock`, but if I had the outer variable also named `clock`, this would get confusing.